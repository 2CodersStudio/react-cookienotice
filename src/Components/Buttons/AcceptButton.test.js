// @flow

import * as React from 'react'
import { shallow } from 'enzyme'
import AcceptButton from './AcceptButton'

describe('AcceptButton', () => {
  describe('without darkTheme', () => {
    const comp = (
      <AcceptButton label='label' onButtonClick={() => {}} darkTheme={false} />
    )

    it('should render', () => {
      expect(comp).toMatchSnapshot()
    })

    it('should have default class', () => {
      const wrapper = shallow(comp)
      expect(wrapper.find('.react-cookienotice-accept-button')).toHaveLength(1)
    })

    it('should not have dark class', () => {
      const wrapper = shallow(comp)
      expect(wrapper.find('.dark')).toHaveLength(0)
    })
  })

  describe('with darkTheme', () => {
    const comp = (
      <AcceptButton label='label' onButtonClick={() => {}} darkTheme />
    )

    it('should render', () => {
      expect(comp).toMatchSnapshot()
    })

    it('should have default class', () => {
      const wrapper = shallow(comp)
      expect(wrapper.find('.react-cookienotice-accept-button')).toHaveLength(1)
    })

    it('should have dark class', () => {
      const wrapper = shallow(comp)
      expect(wrapper.find('.dark')).toHaveLength(1)
    })
  })

  it('should handle click', () => {
    const onClick = jest.fn()

    const wrapper = shallow(
      <AcceptButton label='label' onButtonClick={onClick} darkTheme={false} />
    )

    wrapper.simulate('click')

    expect(onClick).toHaveBeenCalled()
  })
})
