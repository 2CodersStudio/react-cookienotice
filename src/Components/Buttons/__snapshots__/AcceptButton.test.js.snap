// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcceptButton with darkTheme should render 1`] = `
<AcceptButton
  darkTheme={true}
  label="label"
  onButtonClick={[Function]}
/>
`;

exports[`AcceptButton without darkTheme should render 1`] = `
<AcceptButton
  darkTheme={false}
  label="label"
  onButtonClick={[Function]}
/>
`;
